packages/cli-tools/src/__tests__/resolveTransitiveDeps.test.ts

```diff
@@ -0,0 +1,267 @@
+      ['*', '>=2.2.0', '>=2.0.0'],
+      ['1.9.0', '2.0.0', '2.2.0', '3.0.0'],
+    );
+
+    expect(workingVersion).toBe('3.0.0');
+  });
+
+  it('should return null if no version matches all ranges', () => {
+    const workingVersion = calculateWorkingVersion(
+      ['*', '>=2.2.0', '^1.0.0-alpha'],
+      ['1.9.0', '2.0.0', '2.1.0'],
+    );
+
+    expect(workingVersion).toBe(null);
+  });
+});
+
+describe('findDependencyPath', () => {
+  it('should return the path to the dependency if it is in top-level node_modules', () => {
+    writeFiles(DIR, {
+      'package.json': JSON.stringify(rootPackageJson),
+      'node_modules/@react-navigation/stack/package.json': JSON.stringify(
+        stackPackageJson,
+      ),
+    });
+
+    const dependencyPath = findDependencyPath(
+      '@react-navigation/stack',
+      DIR,
+      path.join(DIR, 'node_modules', '@react-navigation/stack'),
+    );
+
+    expect(dependencyPath).toBe(
+      path.join(DIR, 'node_modules', '@react-navigation/stack'),
+    );
+  });
+
+  it('should return the path to the nested node_modules if package is installed here', () => {
+    writeFiles(DIR, {
+      'package.json': JSON.stringify(rootPackageJson),
+      'node_modules/@react-navigation/stack/node_modules/react-native-gesture-handler/package.json':
+        '{}',
+    });
+
+    const dependencyPath = findDependencyPath(
+      'react-native-gesture-handler',
+      DIR,
+      path.join(DIR, 'node_modules', '@react-navigation/stack'),
+    );
+
+    expect(dependencyPath).toBe(
+      path.join(
+        DIR,
+        'node_modules',
+        '@react-navigation/stack',
+        'node_modules',
+        'react-native-gesture-handler',
+      ),
+    );
+  });
+});
+
+describe('collectDependencies', () => {
+  beforeEach(() => {
+    createTempFiles();
+  });
+
+  it('should recursively get all dependencies', () => {
+    const dependencies = collectDependencies(DIR);
+
+    expect(dependencies.size).toBe(4);
+  });
+
+  it('should collect peer dependencies of a dependency', () => {
+    const dependencies = collectDependencies(DIR);
+    const stackDependency = dependencies.get(
+      '@react-navigation/stack',
+    ) as DependencyData;
+    const peerNames = Object.keys(stackDependency.peerDependencies);
+
+    expect(peerNames).toContain('react');
+    expect(peerNames).toContain('react-native');
+    expect(peerNames).toContain('react-native-gesture-handler');
+  });
+});
+
+describe('filterNativeDependencies', () => {
+  it('should return only dependencies with peer dependencies containing native code', () => {
+    createTempFiles({
+      'node_modules/react-native-safe-area-view/ios/Podfile': '{}',
+    });
+    const dependencies = collectDependencies(DIR);
+    const filtered = filterNativeDependencies(DIR, dependencies);
+    expect(filtered.keys()).toContain('@react-navigation/stack');
+    expect(filtered.keys()).toContain('@react-navigation/elements');
```

Comment:
I don't get what this test verifies. What's `react-native-safe-area-view` changing? Are those all the keys available in `filtered` or only a few?

Reply:
Removing check for `'@react-navigation/stack'` as its not relevant here. `react-native-safe-area-view` is a mocked dependency with native code, which is a peer dependency of `@react-native/elements`, and `@react-native/elements` is a dependency of `@react-navigation/stack`. This test proves that recursively collected packages dependent on peer deps with native code will be listed